src/openzeppelin/token/erc1155/library.cairo

```diff
@@ -0,0 +1,600 @@
+
+        # Add to receiver
+        let (to_balance : Uint256) = ERC1155_balances.read(id=id, account=to)
+        with_attr error_message("ERC1155: balance overflow"):
+            let (new_balance : Uint256) = uint256_checked_add(to_balance, amount)
+        end
+        ERC1155_balances.write(id=id, account=to, value=new_balance)
+
+        let (operator) = get_caller_address()
+
+        TransferSingle.emit(operator, from_, to, id, amount)
+
+        _do_safe_transfer_acceptance_check(operator, from_, to, id, amount, data_len, data)
+
+        return ()
+    end
+
+    func _safe_batch_transfer_from{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        from_ : felt,
+        to : felt,
+        ids_len : felt,
+        ids : Uint256*,
+        amounts_len : felt,
+        amounts : Uint256*,
+        data_len : felt,
+        data : felt*,
+    ):
+        alloc_locals
+        with_attr error_message("ERC1155: transfer to the zero address"):
+            assert_not_zero(to)
+        end
+        # Check args are equal length arrays
+        with_attr error_message("ERC1155: ids and amounts length mismatch"):
+            assert ids_len = amounts_len
+        end
+        # Recursive call
+        let len = ids_len
+        safe_batch_transfer_from_iter(from_, to, len, ids, amounts)
+        let (operator) = get_caller_address()
+        TransferBatch.emit(operator, from_, to, ids_len, ids, amounts_len, amounts)
+        _do_safe_batch_transfer_acceptance_check(
+            operator, from_, to, ids_len, ids, amounts_len, amounts, data_len, data
+        )
+        return ()
+    end
+
+    func _mint{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        to : felt, id : Uint256, amount : Uint256, data_len : felt, data : felt*
+    ):
+        let (caller) = get_caller_address()
+        with_attr error_message("ERC1155: called from zero address"):
+            assert_not_zero(caller)
+        end
+        # Cannot mint to zero address
+        with_attr error_message("ERC1155: mint to the zero address"):
+            assert_not_zero(to)
+        end
+        # Check uints valid
+        with_attr error_message("ERC1155: id is not a valid Uint256"):
+            uint256_check(id)
+        end
+        with_attr error_message("ERC1155: amount is not a valid Uint256"):
+            uint256_check(amount)
+        end
+
+        # add to minter check for overflow
+        let (to_balance : Uint256) = ERC1155_balances.read(id=id, account=to)
+        with_attr error_message("ERC1155: balance overflow"):
+            let (new_balance : Uint256) = uint256_checked_add(to_balance, amount)
+        end
+        ERC1155_balances.write(id=id, account=to, value=new_balance)
+
+        let (operator) = get_caller_address()
+        TransferSingle.emit(operator=operator, from_=0, to=to, id=id, value=amount)
+        _do_safe_transfer_acceptance_check(operator, 0, to, id, amount, data_len, data)
```

Comment:
I think these were autogenerated by cairo-format, and only crop up in events and storage operations. I like their usage when the arguments of the called function aren't obvious without scrolling to the definition, but I don't think that I manually included any

Reply:
Ahhh that makes sense. Autogeneration is definitely super convenient when coding. It's not the worst problem to have, I'm just looking at it from a formatting standpoint. Maybe as a loose rule: if there's 5+ params or it's not obvious, let's add kwargs. Thoughts?

Reply:
I guess this is a comment on a pretty old point, I mostly agree, but I think this should be refined to "if there are 5+ params and the names of the inputs don't correspond exactly to those params, or it's otherwise not obvious then use kwargs." Since a lot of the time there are 5+ params but they do correspond exactly, and there is no ambiguity.

Reply:
Yeah, that's totally reasonable. If the inputs don't exactly correspond, I'm on board to include them.